### PR TITLE
Update collector core and contrib to latest and fix example components usage

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: internal/receiver/smartagentreceiver/output_test.go [hunk 1]
```diff
@@ -264,7 +264,18 @@ type mockMetadataClient struct {
...
+func (me *mockMetadataClient) Start(_ context.Context, _ component.Host) error {
+	panic("implement me")
+}
+
+func (me *mockMetadataClient) Shutdown(_ context.Context) error {
+	panic("implement me")
+}
+
+func (me *mockMetadataClient) ConsumeMetrics(_ context.Context, _ pdata.Metrics) error {
+	panic("implement me")
```
Comment (bogdandrutu) [inline internal/receiver/smartagentreceiver/output_test.go:278]: If anyone can file an issue in the core about this use-case to be provided/facilitate in core would be appreciated.

**Review by dmitryax — APPROVED** [review]

File: internal/receiver/smartagentreceiver/output_test.go [hunk 2]
```diff
@@ -158,7 +158,7 @@ func TestSendDimensionUpdateFromConfigMetadataExporters(t *testing.T) {
 			DimensionClients: []string{"mockmetadataexporter", "exampleexporter", "metricsreceiver", "notareceiver", "notreal"},
 		},
 		fakeMonitorFiltering(),
-		&componenttest.ExampleExporterConsumer{},
+		consumertest.NewMetricsNop(),
```
Comment (bogdandrutu) [inline internal/receiver/smartagentreceiver/output_test.go:161]: Here and the next 3 similar usages are just wrong usages of the exporter example (one of the reason to be removed)

File: internal/receiver/smartagentreceiver/output_test.go (hunk 1 above)
Comment (bogdandrutu) [reply internal/receiver/smartagentreceiver/output_test.go:278]: Also an investigation if this is required at all to implement that interface will be good

File: internal/receiver/smartagentreceiver/output_test.go (hunk 2 above)
Comment (rmfitzpatrick) [reply internal/receiver/smartagentreceiver/output_test.go:161]: What is wrong with its usage?  This test exercises usage of incompatible components and confirms they don't conflict with valid ones.

**Pull request merged**